### PR TITLE
Documentation for image.center_crop api.

### PR DESCRIPTION
**Pull request by pracheer**
In order to figure out the possible interpolation techniques, I referred to http://docs.opencv.org/2.4/modules/imgproc/doc/geometric_transformations.html#void%20resize(InputArray%20src,%20OutputArray%20dst,%20Size%20dsize,%20double%20fx,%20double%20fy,%20int%20interpolation)

@mli @zackchase @madjam @nswamy

### CODE REVIEW

File: python/mxnet/image.py [hunk 1]
```diff
@@ -101,7 +101,57 @@ def random_crop(src, size, interp=2):
 
 
 def center_crop(src, size, interp=2):
-    """Centrally crop src with size. Upsample result if src is smaller than size."""
+    """Crops the image `src` to the given `size` by trimming all the four sides
+    while retaining the center of the image. Upsamples if `src` is smaller than
+    `size`.
+
+    Note: This requires OpenCV to be installed.
```
Comment (piiswrong) [inline python/mxnet/image.py]: requires mxnet to be compiled with USE_OPENCV.
also this not the right format for Note

File: python/mxnet/image.py [hunk 2]
```diff
@@ -101,7 +101,57 @@ def random_crop(src, size, interp=2):
...
+
+    Parameters
+    ----------
+    src : NDArray
+        Binary image data as `NDArray`.
+    size : list or tuple of int
+        The putput image size.
```
Comment (piiswrong) [inline python/mxnet/image.py]: output

File: python/mxnet/image.py [hunk 3]
```diff
@@ -101,7 +101,57 @@ def random_crop(src, size, interp=2):
...
+
+    Example
+    -------
+    >>> with open("flower.jpg", 'rb') as fp:
+    ...     str_image = fp.read()
+    ...
+    >>> image = mx.image.imdecode(str_image)
+    >>> image
+    <NDArray 2321x3482x3 @cpu(0)>
+    >>> cropped_image = mx.image.center_crop(image, (1000, 500))
```
Comment (piiswrong) [inline python/mxnet/image.py]: cropped_image, (x, y, width, height) =

File: python/mxnet/image.py [hunk 4]
```diff
@@ -101,7 +101,57 @@ def random_crop(src, size, interp=2):
...
+        4: Lanczos interpolation over 8x8 pixel neighborhood.
+
+         To shrink an image, it will generally look best with AREA-based
+        interpolation, whereas to enlarge an image, it will generally look best
+        with Bicubic (slow) or Bilinear (faster but still looks OK).
+
+    Returns
+    -------
+    NDArray
+        The cropped image.
```
Comment (piiswrong) [inline python/mxnet/image.py]: also returns a tuple describing the position of crop

File: python/mxnet/image.py [hunk 5]
```diff
@@ -101,7 +101,63 @@ def random_crop(src, size, interp=2):
 
 
 def center_crop(src, size, interp=2):
-    """Centrally crop src with size. Upsample result if src is smaller than size."""
+    """Crops the image `src` to the given `size` by trimming on all the four
```
Comment (zackchase) [inline python/mxnet/image.py]: remove "the" -> "trimming on all four sides"

File: python/mxnet/image.py [hunk 6]
```diff
@@ -101,7 +101,63 @@ def random_crop(src, size, interp=2):
...
+        2: Area-based (resampling using pixel area relation). It may be a
+        preferred method for image decimation, as it gives moire-free
+        results. But when the image is zoomed, it is similar to the Nearest
+        Neighbors method. (used by default).
+
+        3: Bicubic interpolation over 4x4 pixel neighborhood.
+
+        4: Lanczos interpolation over 8x8 pixel neighborhood.
+
+         To shrink an image, it will generally look best with AREA-based
```
Comment (zackchase) [inline python/mxnet/image.py]: change to-> "When shrinking an image, ..."

File: python/mxnet/image.py [hunk 7]
```diff
@@ -101,7 +101,63 @@ def random_crop(src, size, interp=2):
...
+        interpolation, whereas to enlarge an image, it will generally look best
```
Comment (zackchase) [inline python/mxnet/image.py]: -> "whereas, when enlarging an image"

**Pull request merged**